lib/consumer.js

```diff
@@ -92,50 +114,53 @@ var checkRpc = function(msg, queue) {
   };
 };
 
-var consume = function(queue, options, callback) {
+var consume = (queue, options, callback) => {
   if (typeof options === 'function') {
     callback = options;
     options = { persistent: true, durable: true };
   }
 
-  utils.pushIfNotExist('consumer', amqpConsumptions, { queue: queue, options: options, callback: callback });
-
-  if (amqpState !== amqpStates.CONNECTED) {
-    return connect()
-    .then(function() {
-      return consume(queue, options, callback);
-    });
-  }
-
   queue += (process.env.LOCAL_QUEUE) ? process.env.LOCAL_QUEUE : '';
 
-  return amqpChannel.assertQueue(queue, options)
-  .then(function (_q) {
-    Logger.info('[BMQ-CONSUMER] Consume from queue:', _q.queue);
-    amqpChannel.consume(_q.queue, function (_msg) {
-      Logger.info('[BMQ-CONSUMER][' + _q.queue + '] < ' + _msg.content.toString());
-
-      Promise.resolve(parsers.in(_msg))
-      .then(callback)
-      .then(checkRpc(_msg, _q.queue))
-      .then(function () {
-        amqpChannel.ack(_msg);
-      })
-      .catch(function (_err) {
-        Logger.error('[BMQ-CONSUMER] Error on queue: ' + _q.queue, _err);
-        amqpChannel.reject(_msg, amqpConfig.amqpRequeue);
-      });
-    }, { noAck: false });
 
-    return true;
+  return Promise.resolve()
+  .then(connect)
+  .then(createChannel)
+  .then(() => {
+    return amqpChannel.assertQueue(queue, options)
+    .then((_queue) => {
+      utils.pushIfNotExist(amqpQueues, { queue: queue, options: options, callback: callback });
+
+      Logger.info('[BMQ-CONSUMER] Consume from queue:', _queue.queue);
+      amqpChannel.consume(_queue.queue, (_msg) => {
+        Logger.info('[BMQ-CONSUMER][' + _queue.queue + '] < ' + _msg.content.toString());
+
+        Promise.resolve(parsers.in(_msg))
+        .then(callback)
+        .then(checkRpc(_msg, _queue.queue))
```

Comment:
Why not? It's just chaining, how would you do otherwise? Can you provide a simple sample based on this 2 calls?